service/remote.go

```diff
@@ -140,24 +141,46 @@ func (r *RemoteService) AddRemoteBindingListener(bindingListener cluster.RemoteB
 func (r *RemoteService) Call(ctx context.Context, req *protos.Request) (*protos.Response, error) {
 	c, err := util.GetContextFromRequest(req, r.server.ID)
 	c = util.StartSpanFromRequest(c, r.server.ID, req.GetMsg().GetRoute())
+	defer tracing.FinishSpan(c, err)
 	var res *protos.Response
+
+	if err == nil {
+		result := make(chan *protos.Response, 1)
+		go func() {
+			result <- processRemoteMessage(c, req, r)
+		}()
+
+		// Set a timeout for processing the call
+		timeout := time.Duration(300) * time.Second
```

Comment:
Why 300 seconds? How did you come up with this number? Lets make it configurable and use a smaller time :)

Reply:
This is just a safety net to ensure there is a set value (in case backend is updated and frontend is not), this is also the existing hardcoded default on the properties below. This is already configurable, the developer just needs to set the request timeout with "pitaya.cluster.rpc.client.nats.requesttimeout" or "pitaya.cluster.rpc.client.grpc.requesttimeout".

Reply:
Actually the hardcoded default is 5 seconds, not 5 minutes 😂 in any case, the default on the backend might not be the same as the default on the frontend, hence a higher default is safer. Here's an example:

Reply:
Actually, upon some real life testing, looks like the timeout isn't making its way to the server (metadata isn't being unmarshalled) need to fix this, hold on ;)